rkt/cli_apps.go

```diff
@@ -150,6 +155,89 @@ func (ae *appExec) Set(s string) error {
 	return nil
 }
 
+// appInjectVolume is for --inject-volume flags in the form of: --inject-volume source=PATH,target=PATH
+type appInjectVolume apps.Apps
+
+func (aam *appInjectVolume) Set(s string) error {
+	app := (*apps.Apps)(aam).Last()
+	if app == nil {
+		return fmt.Errorf("--inject-volume must follow an image")
+	}
+
+	p := strings.SplitN(s, ":", 2)
+	if len(p) != 2 {
+		return fmt.Errorf("must be SOURCE_PATH:TARGET_PATH")
+	}
+
+	volName, err := types.SanitizeACName(p[1])
+	if err != nil {
+		return fmt.Errorf("empty target in volume")
+	}
+
+	rVol := types.Volume{Name: *types.MustACName(volName), Source: p[0], Kind: "host"}
+	mount := schema.Mount{Volume: rVol.Name, Path: p[1]}
+
+	(*apps.Apps)(aam).Volumes = append((*apps.Apps)(aam).Volumes, rVol)
+	if (*apps.Apps)(aam).Count() == 0 {
+		(*apps.Apps)(aam).Mounts = append((*apps.Apps)(aam).Mounts, mount)
+	} else {
+		app := (*apps.Apps)(aam).Last()
+		app.Mounts = append(app.Mounts, mount)
+	}
+
+	return nil
+}
+
+func (aam *appInjectVolume) Type() string {
+	return "appInjectVolume"
+}
+
+func (aam *appInjectVolume) String() string {
+	return ""
+}
+
+// appMount is for --mount flags in the form of: --mount volume=VOLNAME,target=PATH
+type appMount apps.Apps
+
+func (al *appMount) Set(s string) error {
+	mount := schema.Mount{}
+
+	// this is intentionally made similar to types.VolumeFromString()
+	// TODO(iaguis) use MakeQueryString() when appc/spec#520 is merged
+	m, err := url.ParseQuery(strings.Replace(s, ",", "&", -1))
+	if err != nil {
+		return err
+	}
+
+	for key, val := range m {
+		if len(val) > 1 {
+			return fmt.Errorf("label %s with multiple values %q", key, val)
+		}
+		switch key {
+		case "volume":
+			mv, err := types.NewACName(val[0])
+			if err != nil {
+				return err
```

Comment:
Please add some context to this error, like `fmt.Errorf("Invalid volume name %q in --mount flag %q: %v", val[0], s, err)`